### PR TITLE
Guozhong/remove testcase load network single ie core

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *remove testcase  smoke_LoadNetwork_SingleIECore*

### Tickets:
 - *ticket-89440*

### CODE REVIEW

File: src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp [hunk 1]
```diff
@@ -333,20 +333,6 @@ TEST_P(CoreThreadingTestsWithIterations, smoke_LoadNetworkAccuracy) {
...
-    std::atomic<unsigned int> counter{0u};
-    InferenceEngine::Core ie;
-
-    SetupNetworks();
-
-    runParallel([&] () {
-        auto value = counter++;
-        ie.SetConfig(config, deviceName);
-        (void)ie.LoadNetwork(networks[value % networks.size()], deviceName);
-    }, numIterations, numThreads);
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:347]: why?

File: src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp [hunk 2]
```diff
@@ -333,20 +333,6 @@ TEST_P(CoreThreadingTestsWithIterations, smoke_LoadNetworkAccuracy) {
     }, numIterations, numThreads);
 }
 
-// tested function: single IECore ReadNetwork, SetConfig, LoadNetwork, AddExtension
-TEST_P(CoreThreadingTestsWithIterations, smoke_LoadNetwork_SingleIECore) {
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:337]: We shouldn't remove test cases, we need to understand why it happens.

**Review by ilyachur — DISMISSED** [review]

File: src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp (hunk 2 above)
Comment (songbell) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:337]: @ilyachur @ilya-lavrenov   please see my comments in jira - 89440, this case is not thread-safe in current plugin implementation.

Comment (ilya-lavrenov) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:337]: It was threads safe for many releases, what was change? In it's a regression, we need to address it

Comment (songbell) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:337]: ilya, this case is newly added(https://github.com/openvinotoolkit/openvino/pull/11796/files), not the one existing for a long time.

Comment (ilya-lavrenov) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:337]: Is the issue seen w/o https://github.com/openvinotoolkit/openvino/pull/12240 ?

Comment (songbell) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:337]: yes, seen w/o #12240, I have found below logs in jira ticket 85384, the date is before @12440 merged. @wgzintel  have we reproduce this failures locally?
ie-tests-linux-ubuntu20-cpu/59462/consoleText-[2022-07-05T01:28:21.642Z] [2022-07-05 01:28:21,469] [139666179897088]
CPU/CoreThreadingTestsWithIterations.smoke_LoadNetwork_SingleIECore/targetDevice=CPU_config=PERF_COUNT:YES_numThreads=4_numIter=50
Killed
--
ie-tests-linux-ubuntu20-cpu/59458/consoleText-[2022-07-05T00:02:15.199Z] [2022-07-05 00:02:14,948] [140252173358848]
CPU/CoreThreadingTestsWithIterations.smoke_LoadNetwork_SingleIECore/targetDevice=CPU_config=PERF_COUNT:YES_numThreads=4_numIter=50
Killed

Comment (wgzintel) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:345]: This failures did not reproduce locally, I have tried to run 500 times, no killed.

Comment (ilya-lavrenov) [reply src/tests/functional/plugin/shared/include/behavior/plugin/core_threading.hpp:345]: discussed offline, the test is invalid, you are right.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
do we need to port to 2022.2 branch?

**Comment by songbell** [comment]
> do we need to port to 2022.2 branch?



> do we need to port to 2022.2 branch?

I think depends on 22.2 development status, if still some development activities there, we'd better port it to avoid these case failures.

**Review by ilyachur — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
> 

let's port if the tests exists in 2022.2

**Pull request merged**